### ISSUE TITLE
[MRTK3] Quest hands don't issue UI Select on their controllers

### ISSUE THREAD

**Issue by akash14darshan** [microsoft/MixedRealityToolkit-Unity#11284]
## Describe the bug

Unity's own canvas doesn't receive input from hands, although the Far ray is still landing on the canvas element.
I assume it could be an unity bug, but still leaving a report here. 
Controllers work as they are supposed to, So does the simulated hand on editor and the normal mouse click.

## To reproduce

1. In the MRTKDevTemplate, create a new scene. 
2. Delete the default camera, and add MRTK XR Rig, EventSystem and MRTKInputSimulator prefabs to scene.
3. Create any UI element, lets say UI Slider
4. Place the slider appropriately infront of camera with right scale (i'm using Follow script for convenience)
5. Add "Tracked Device Graphic Raycaster" component to the canvas.
6. Optionally assign XR Rig's main camera as the event camera
7. Add scene to build options
8. [Configure project for quest2 ](https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/deployment/quest-deployment)
9. Build and test now.

Alternatively, import this unitypackage in MRTKDevTemplate, and follow from step6 
[testMrtk3CanvasUI.zip](https://github.com/microsoft/MixedRealityToolkit-Unity/files/10196303/testMrtk3CanvasUI.zip)

## Expected behavior

I will expect UI to work with both hands and controlleras it does in MRTK2

## Footage

https://user-images.githubusercontent.com/13174425/206743495-67cbb5b7-7af8-4482-86b2-c1e38d656a55.mp4

## Your setup (please complete the following information)

- Unity Version [e.g. 2020.3.42f1]
- MRTK 3 (cloned repo)

## Target platform (please complete the following information)

- Quest 2

**Comment by Zee2** [comment]
Hey there,

Unfortunately we don't currently support near/touch interaction on vanilla UGUI canvases. This is because we entirely rely on XRI's Canvas raycasting implementation for back-compat to vanilla UGUI canvases, and XRI does not support non-ray-based UI interactions.

The new UXComponents are fully Canvas-based, so you can build your UI with all of the familiar RectTransform/Canvas tooling, but also including all of the near-touch/XR input that you'd expect.

In the future, if/when XRI decides to include this functionality, we will make sure that it works with our rig.

**Comment by Zee2** [comment]
https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.2/manual/ui-setup.html

See the XRI docs on UI setup, and note that they only support ray-based interactions.

**Comment by akash14darshan** [comment]
@Zee2 that is indeed far ui interaction. Not a near ui.
You can see the footage for that (far rays are still visible)

**Comment by akash14darshan** [comment]
@Zee2 yes you're totally right. On digging the code, i found some evidences for the same.
https://github.com/microsoft/MixedRealityToolkit-Unity/compare/main...akash14darshan:MixedRealityToolkit:mrtk3_handUIPinch?expand=1

Let me know if this looks right. I can submit a PR to save your time :D

**Comment by Zee2** [comment]
I have a branch open for it, but I haven't PR'd it yet because I've experienced unreliability in the past with triggering main Select and UIPressed simultaneously. I need to do some detailed testing first.

**Comment by Zee2** [comment]
@akash14darshan https://github.com/Zee2/MixedRealityToolkit-Unity/tree/fix_polyfill_uipress you can try giving this a whirl. I haven't gotten around to doing the testing yet but you can take a look and see if anything obvious breaks.

**Comment by akash14darshan** [comment]
@Zee2 With those changes I noticed something odd with the visuals. Although I'm not sure if that is related to the current issue.

https://user-images.githubusercontent.com/13174425/207364125-3c7b2696-b75f-46bb-af78-8ccff57411b1.mp4

If you look at the mrtk cursor (at the point of intersection of ray and canvas) even though I have kept the "pinch" pose, the cursor still goes to a state where it doesn't indicate pinching.

Ideally the cursor should stay "stuck" to the point where it initially intersects with the canvas, and stay there until released. 

To reproduce:
While pinching on some UI element, move your ray to outside the canvas and back into it twice without releasing pinch.
This can also be reproduced on quest.
